src/main/java/org/csanchez/jenkins/plugins/kubernetes/AllContainersRunningPodWatcher.java

```diff
@@ -72,7 +84,29 @@ boolean areAllContainersRunning(Pod pod) {
         }
         for (ContainerStatus containerStatus : containerStatuses) {
             if (containerStatus != null) {
-                if (containerStatus.getState().getWaiting() != null) {
+                ContainerStateWaiting waitingState = containerStatus.getState().getWaiting();
+                if (waitingState != null) {
+                    String waitingStateMsg = waitingState.getMessage();
+                    if (waitingStateMsg != null && waitingStateMsg.contains("Back-off pulling image")) {
+                        runListener.error("Unable to pull Docker image \""+containerStatus.getImage()+"\". Check if image name is spelled correctly");
```

Comment:
This ought to be straightforward to add a test for (probably in `KubernetesPipelineTest`): a script requesting a bogus `image`, start building, verify that the build is aborted, `assertLogContains` this text (preferably factored into a `static` utility method for use from the test) and `org.jenkinsci.plugins.workflow.steps.durable_task.Messages.ExecutorStepExecution_queue_task_cancelled()`.